### PR TITLE
Adds TARDIS

### PR DESCRIPTION
**Pull request by monster860**
It's basically a locker that's bigger inside than outside. Other than that, there's nothing special about it. It starts in the derelict.

:cl: monster860
rscadd: Adds a locker bigger on the inside than on the outside, the TARDIS
rscadd: Actual area you teleport to when entering the tardis is on Z2
/:cl:

### CODE REVIEW

**Comment by tigercat2000** [comment]
http://niceme.me

**Comment by monster860** [comment]
http://http://nicememe.website/

File: code/game/objects/structures/crates_lockers/closets/tardis.dm [hunk 1]
```diff
@@ -0,0 +1,81 @@
...
+	icon_opened = "emergencyopen"
+
+/obj/structure/closet/tardis/New()
+	..()
+	the_tardis = src
+
+/obj/structure/closet/tardis/close()
+	. = ..()
+	var/obj/effect/landmark/W
+	for(var/obj/effect/landmark/L in landmarks_list)
```
Comment (tigercat2000) [inline code/game/objects/structures/crates_lockers/closets/tardis.dm]: you could just use the /area/tardis area along with one of the "safe teleport" procs to move mobs there

File: code/game/objects/structures/crates_lockers/closets/tardis.dm [hunk 2]
```diff
@@ -0,0 +1,81 @@
...
+		if(L.name == "tardistele")
+			W = L
+			break
+	
+	for(var/atom/movable/A as obj|mob in src)
+		A.forceMove(get_turf(W))
+		if(ismob(A))
+			var/mob/M = A
+			M.client.eye = M.client.mob
```
Comment (tigercat2000) [inline code/game/objects/structures/crates_lockers/closets/tardis.dm]: don't do this or the below line, call M.reset_eye()

File: code/game/objects/structures/crates_lockers/closets/tardis.dm (hunk 1 above)
Comment (monster860) [inline code/game/objects/structures/crates_lockers/closets/tardis.dm]: *walks into tardis*
*ends up on the other side of the entrance*

Yeaaaaaaaahhh..... No.

**Comment by CrAzYPiLoT-SS13** [comment]
Ooh, Travis is complaining

**Comment by NTSAM** [comment]
Portable engiforts?

File: code/game/objects/structures/crates_lockers/closets/tardis.dm (hunk 2 above)
Comment (monster860) [inline code/game/objects/structures/crates_lockers/closets/tardis.dm]: There's no reset_eye

**Comment by Aurorablade** [comment]
For various reasons besides copyright. No.
For various reason including copyright. No.

File: code/game/objects/structures/crates_lockers/closets/tardis.dm (hunk 2 above)
Comment (tigercat2000) [inline code/game/objects/structures/crates_lockers/closets/tardis.dm]: it's `reset_view`

**Comment by tigercat2000** [comment]
@Aurorablade copyright law is not invokable on anything but the stolen name

**Comment by fludd12** [comment]
I like the idea of bluespace pocket-lockers. It'd be... Interesting to make more than one or two, though.

**Comment by HereticChurch** [comment]
Call it the BARDIS and it's fine.
Bluespace And Relative Dimensions In Space.
:+1:

**Comment by CrAzYPiLoT-SS13** [comment]
Make another TARDIS inside a TARDIS and continue going on. What happens? Noone knows.

**Comment by fludd12** [comment]
How would it even work to try and create more than one? Would there be a Z-level just for these things, or would it just keep generating more Z-levels or something?

**Comment by FlattestGuitar** [comment]
The tardis room is on z2,  not really possible to make more than one. 

You can also weld the locker shut trapping anyone inside. 

Thanks to monster's beautiful demonstration.

**Comment by QuinnAggeler** [comment]
I'd be for it, provided there's only one. Possibly with a rename.

**Comment by Fox-McCloud** [comment]
Inclined to agree with @Aurorablade.

**Comment by Spacemanspark** [comment]
If we want to talk about copyright, why do we have xenomorphs?
And what other reasons are there besides copyright? You can't just say "I don't like this for various reasons" and not tell us why.

**Comment by Aurorablade** [comment]
okay heres a list of reasons, my apologies:
While yes we do have engiforts, at least engie forts can be easily sometimes found, this is a bluespace pocket mapped into the game and not admin spawned in a direlict location not easily gotten to outside a space pod. It has a potional for rampant abuse (Hacked Upload) and general bad behavior by trying to exploit mechnics. 

You could pretty much just do something as an antag or non antag that could hold up a round/be used for an objective/hide a body, build a wall around this locker, and hide yourself/something in it for the rest of the round.
That is NOT the type of gameplay i would like to see. If it was something not mapped in and ADMIN SPAWN only or in a MORE ACCESSIBLE location i would go with it. But it has alot of potional for ass hattery here.

**Comment by Spacemanspark** [comment]
@Aurorablade Thank you for providing an answer.
As to your edit- those walls can't be tampered with, I do believe, not without admin abuse.

**Comment by Aurorablade** [comment]
A resourceful person could find a way around it..you have seen what some people on this server are capable of.

**Comment by Spacemanspark** [comment]
I highly doubt it, honestly. Teleports outside of admin versions don't work on the CC Z-level. Wizards can't jaunt out of it, hand teleporters shouldn't work in it, RCDs won't work with it, etc..

**Comment by ProperPants** [comment]
There is a wall that is literally invincible. @Crazylemon64 spawned them once when they made some ghosts into wizards and gave us all firebolt staffs. We destroyed all the walls of the huge empty admin room, then Lemon replaced them with invici-walls.

**Comment by TheDZD** [comment]
I don't see a reason to add yet another safe place for people to hide at, especially one on Z2, with only one way to get to it. Also adds yet another trademark infringing thing to the codebase for inane reasons.

**Comment by monster860** [comment]
How about this for a plan:

- Give the T.A.R.D.I.S. nuke-disk behavior (If destroyed, respawn in random place)
- Make any tracking devices (GPS, suit sensors, etc) show the TARDIS's location
- Name change if needed

**Comment by tigercat2000** [comment]
How about this for a plan:
 - not adding it

**Comment by monster860** [comment]
With the above mentioned changes, it is basically a glorified nuke disk locker. What's wrong with that?

**Comment by tigercat2000** [comment]
what *isn't* wrong with that

**Comment by Fox-McCloud** [comment]
I concur with Tiger and DZD's sentiments here.

**Comment by monster860** [comment]
>in a direlict location not easily gotten to outside a space pod

I go to the derelict all the time without a spacepod. Hell, I've gotten there as a cluwne many times!

**Comment by Fox-McCloud** [comment]
For reasons stated above, and the fact of...`memes`, well, yeah. closing this.

**Pull request closed without merging**